rfaas/lib/executor.cpp

```diff
@@ -34,63 +34,92 @@ namespace rfaas {
     return _timeout;
   }
 
-  executor_state::executor_state(rdmalib::Connection* conn, int rcv_buf_size):
-    conn(conn),
-    _rcv_buffer(rcv_buf_size)
+  libfabric_executor::libfabric_executor(std::string address, int port, int rcv_buf_size, int max_inlined_msg):
+    executor(address, port, rcv_buf_size, max_inlined_msg)
   {
+    _invoc_id = 1;
+    _execs_buf.register_memory(_state.pd(), FI_WRITE | FI_REMOTE_WRITE);
   }
 
-  executor::executor(std::string address, int port, int rcv_buf_size, int max_inlined_msg):
-    _state(address, port, rcv_buf_size + 1),
-    _rcv_buffer(rcv_buf_size),
-    _execs_buf(MAX_REMOTE_WORKERS),
-    _address(address),
-    _port(port),
-    _rcv_buf_size(rcv_buf_size),
-    _executions(0),
-    #ifdef USE_LIBFABRIC
-    _invoc_id(1),
-    #else
-    _invoc_id(0),
-    #endif
-    _max_inlined_msg(max_inlined_msg),
-    _perf(1000)
+  verbs_executor::verbs_executor(std::string address, int port, int rcv_buf_size, int max_inlined_msg):
+    executor(address, port, rcv_buf_size, max_inlined_msg)
   {
-    #ifdef USE_LIBFABRIC
-    _execs_buf.register_memory(_state.pd(), FI_WRITE | FI_REMOTE_WRITE);
-    #else
+    _invoc_id = 0;
     _execs_buf.register_memory(_state.pd(), IBV_ACCESS_LOCAL_WRITE | IBV_ACCESS_REMOTE_WRITE);
-    #endif
-    events = 0;
-    _active_polling = false;
-    _end_requested = false;
   }
 
-  executor::executor(device_data & dev):
-    executor(dev.ip_address, dev.port, dev.default_receive_buffer_size, dev.max_inline_data)
-  {}
-
-  executor::~executor()
+  rdmalib::Buffer<char, libfabric> libfabric_executor::load_library(std::string path)
```

Comment:
Is this function any different from ibverbs implementation?

Reply:
Well, it is different in the sense that they return different kinds of buffers. I guess I can make it return the parent buffer class?

Reply:
Okay, in that context it's fine - unless you can just make it a template in a parent class (depends if anything changes beyond the type itself).